### PR TITLE
New: Translate settings pages

### PR DESCRIPTION
**Pull request by stevietv**
#### Database Migration
NO

#### Description
Implementing translations in the frontend settings pages

This relies on #5847 being merged first

#### Todos
- [x] Settings page
- [x] Media Management
- [x] Profiles
- [x] Quality
- [x] Custom Formats
- [x] Indexers
- [x] Download Clients
- [x] Import Lists
- [x] Connect
- [x] Metadata
- [x] Metadata Source
- [x] Tags
- [x] General
- [x] UI

#### Issues Fixed or Closed by this PR

*

### CODE REVIEW

File: frontend/src/Settings/Tags/AutoTagging/Specifications/AddSpecificationModalContent.js [hunk 1]
```diff
@@ -66,7 +67,7 @@ export default function AddSpecificationModalContent(props) {
 
               <Alert kind={kinds.INFO}>
                 <div>
-                  {'Sonarr supports the follow properties for auto tagging rules'}
+                  {translate('SonarrSupportsTheFollowingPropertiesForAutoTaggingRules')}
```
Comment (markus101) [inline frontend/src/Settings/Tags/AutoTagging/Specifications/AddSpecificationModalContent.js:70]: Similar thing here.

File: frontend/src/Settings/Tags/AutoTagging/AutoTaggings.js [hunk 2]
```diff
@@ -55,9 +56,9 @@ export default function AutoTaggings() {
   }, [dispatch]);
 
   return (
-    <FieldSet legend="Auto Tagging">
+    <FieldSet legend={translate('AutoTagging')}>
       <PageSectionContent
-        errorMessage="Unable to load auto taggimg"
+        errorMessage={translate('UnableToLoadAutoTagging')}
```
Comment (markus101) [inline frontend/src/Settings/Tags/AutoTagging/AutoTaggings.js:61]: `AutoTaggingLoadError` maybe?

File: frontend/src/Settings/Quality/Definition/QualityDefinitions.js [hunk 3]
```diff
@@ -18,20 +19,26 @@ class QualityDefinitions extends Component {
     } = this.props;
 
     return (
-      <FieldSet legend="Quality Definitions">
+      <FieldSet legend={translate('QualityDefinitions')}>
         <PageSectionContent
-          errorMessage="Unable to load Quality Definitions"
+          errorMessage={translate('UnableToLoadQualityDefinitions')}
```
Comment (markus101) [inline frontend/src/Settings/Quality/Definition/QualityDefinitions.js:24]: `QualityDefinitionsLoadError`?

File: frontend/src/Settings/Quality/Definition/QualityDefinition.js [hunk 4]
```diff
@@ -231,11 +232,11 @@ class QualityDefinition extends Component {
                 anchor={
                   <Label kind={kinds.INFO}>{minSixty}</Label>
                 }
-                title="Minimum Limits"
+                title={translate('MinimumLimits')}
                 body={
                   <QualityDefinitionLimits
                     bytes={minBytes}
-                    message="No minimum for any runtime"
+                    message={translate('NoMinimumForAnyRuntime')}
```
Comment (markus101) [inline frontend/src/Settings/Quality/Definition/QualityDefinition.js:239]: This one seems alright despite being a literal conversion.

**Review by markus101 — CHANGES_REQUESTED** [review]
Commented on a few strings that are direct conversions from the english string to a key, I know these came from downstream, but we can improve them and downstream will sync up.

File: frontend/src/Settings/Tags/AutoTagging/Specifications/AddSpecificationModalContent.js [hunk 5]
```diff
@@ -55,7 +56,7 @@ export default function AddSpecificationModalContent(props) {
         {
           !isSchemaFetching && !!schemaError ?
             <div>
-              {'Unable to add a new condition, please try again.'}
+              {translate('UnableToAddANewConditionPleaseTryAgain')}
```
[the remark this answers is not in the document]
Comment (stevietv) [reply frontend/src/Settings/Tags/AutoTagging/Specifications/AddSpecificationModalContent.js:59]: replaced all these with less verbose key

File: frontend/src/Settings/Tags/AutoTagging/Specifications/AddSpecificationModalContent.js (hunk 1 above)
Comment (stevietv) [reply frontend/src/Settings/Tags/AutoTagging/Specifications/AddSpecificationModalContent.js:70]: done

File: frontend/src/Settings/CustomFormats/CustomFormats/Specifications/EditSpecificationModalContent.js [hunk 6]
```diff
@@ -40,30 +41,31 @@ function EditSpecificationModalContent(props) {
...
           {...otherProps}
         >
           {
-            fields && fields.some((x) => x.label === 'Regular Expression') &&
+            fields && fields.some((x) => x.label === translate('RegularExpression')) &&
               <Alert kind={kinds.INFO}>
                 <div>
-                  <div dangerouslySetInnerHTML={{ __html: 'This condition matches using Regular Expressions. Note that the characters <code>\\^$.|?*+()[{</code> have special meanings and need escaping with a <code>\\</code>' }} />
-                  {'More details'} <Link to="https://www.regular-expressions.info/tutorial.html">{'Here'}</Link>
+                  <InlineMarkdown data={translate('ThisConditionMatchesUsingRegularExpressions')} />
```
Comment (markus101) [inline frontend/src/Settings/CustomFormats/CustomFormats/Specifications/EditSpecificationModalContent.js:55]: This should also be less verbose.

File: frontend/src/Settings/General/GeneralSettings.js [hunk 7]
```diff
@@ -183,12 +186,10 @@ class GeneralSettings extends Component {
...
           isOpen={this.state.isRestartRequiredModalOpen}
           kind={kinds.DANGER}
-          title="Restart Sonarr"
-          message={
-            `Sonarr requires a restart to apply changes, do you want to restart now? ${isWindowsService ? 'Depending which user is running the Sonarr service you may need to restart Sonarr as admin once before the service will start automatically.' : ''}`
-          }
-          cancelLabel="I'll restart later"
-          confirmLabel="Restart Now"
+          title={translate('RestartSonarr')}
+          message={`${translate('RestartRequiredToApplyChanges')} ${isWindowsService ? translate('RestartRequiredToApplyChangesIsWindowsService') : ''}`}
```
Comment (markus101) [inline frontend/src/Settings/General/GeneralSettings.js:190]: `RestartRequiredToApplyChangesIsWindowsService` this also seems needlessly verbose

File: frontend/src/Settings/General/GeneralSettings.js [hunk 8]
```diff
@@ -183,12 +186,10 @@ class GeneralSettings extends Component {
...
+          cancelLabel={translate('IllRestartLater')}
```
Comment (markus101) [inline frontend/src/Settings/General/GeneralSettings.js:191]: `RestartLater` should be enough.

File: frontend/src/Settings/ImportLists/ImportListExclusions/EditImportListExclusionModalContent.js [hunk 9]
```diff
@@ -58,24 +61,24 @@ function EditImportListExclusionModalContent(props) {
...
-                  helpText="The name of the series to exclude"
+                  helpText={translate('SeriesTitleToExcludeHelpText')}
                   {...title}
                   onChange={onInputChange}
                 />
               </FormGroup>
 
               <FormGroup>
-                <FormLabel>TVDB ID</FormLabel>
+                <FormLabel>{translate('TVDBID')}</FormLabel>
```
Comment (markus101) [inline frontend/src/Settings/ImportLists/ImportListExclusions/EditImportListExclusionModalContent.js:76]: `TvdbId`

File: frontend/src/Settings/ImportLists/ImportListExclusions/EditImportListExclusionModalContent.js [hunk 10]
```diff
@@ -58,24 +61,24 @@ function EditImportListExclusionModalContent(props) {
...
 
                 <FormInputGroup
                   type={inputTypes.TEXT}
                   name="tvdbId"
-                  helpText="The TVDB ID of the series to exclude"
+                  helpText={translate('TVDBIDExcludeHelpText')}
```
Comment (markus101) [inline frontend/src/Settings/ImportLists/ImportListExclusions/EditImportListExclusionModalContent.js:81]: `TvdbId`

File: frontend/src/Settings/Indexers/Indexers/EditIndexerModalContent.js [hunk 11]
```diff
@@ -84,41 +87,41 @@ function EditIndexerModalContent(props) {
               </FormGroup>
 
               <FormGroup>
-                <FormLabel>Enable RSS</FormLabel>
+                <FormLabel>{translate('EnableRSS')}</FormLabel>
```
Comment (markus101) [inline frontend/src/Settings/Indexers/Indexers/EditIndexerModalContent.js:90]: Rss (same for other uses of `RSS` here)

File: frontend/src/Settings/Indexers/Options/IndexerOptions.js [hunk 12]
```diff
@@ -80,16 +83,16 @@ function IndexerOptions(props) {
               advancedSettings={advancedSettings}
               isAdvanced={true}
             >
-              <FormLabel>RSS Sync Interval</FormLabel>
+              <FormLabel>{translate('RSSSyncInterval')}</FormLabel>
```
Comment (markus101) [inline frontend/src/Settings/Indexers/Options/IndexerOptions.js:86]: Rss (and others in this file/elsewhere)

**Review by markus101 — CHANGES_REQUESTED** [review]
There are a few places where we're using capitals for the full acronym instead of following the typical CamelCase naming that we should adjust.

File: frontend/src/Settings/Indexers/Options/IndexerOptions.js (hunk 12 above)
Comment (stevietv) [reply frontend/src/Settings/Indexers/Options/IndexerOptions.js:86]: captured RSS, SSL, UTC, URL and TVDB

**Comment by stevietv** [comment]
> There are a few places where we're using capitals for the full acronym instead of following the typical CamelCase naming that we should adjust.

Done another pass - the only ones I didn't adjust to camelCase were UI - I believe this one stays capitals as its two letters?

**Comment by markus101** [comment]
I think we should do `UI` -> `Ui` as well for consistency, then just needs a rebase.

**Comment by stevietv** [comment]
UI has been renamed to Ui.
Rebased and translations rechecked.

**Comment by markus101** [comment]
Thanks @stevietv!!

**Pull request merged**